bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -553,7 +553,7 @@ impl<'a> CircuitInputStateRef<'a> {
             sender_balance
         );
         // If receiver doesn't exist, create it
-        if (!receiver_exists && !value.is_zero()) || must_create {
+        if !receiver_exists && (!value.is_zero() || must_create) {
```

Comment:
This comment is not a request for change but a note for other reviewers:

I wrapped my head around this expression and finally figured out a more straightforward way to convince myself the proposed change is correct. 

- If the expression evaluates true, we create an account with an empty hash. Specifically, update the codehash from 0 to the EMPTY_CODE_HASH.
- In the previous expression, the `must_create` dominates the outcome. If the call is CREATE, then even if the receiver_exists, the receiver account is still created with the empty hash. 
- In the proposed expression, the `!receiver_exists` dominates the outcome. If the receiver already exists, there's no way to mess up the account code hash again. This is the behavior that makes sense.

The failing case is captured in the test case `create_tx_for_existing_account`.

I've included my previous elaborate boolean analysis in the "Details".


<details><summary>Details</summary>
<p>

Let A = (!receiver_exists), B = !value.is_zero(), and C = must_create,
- before: (A && B) || C
- after: A && (B || C)

The two expressions disagree on the following cases
- A = False,  B = False, C = True
- A = False,  B = True, C = True

In both cases, the receiver exists, and the call is CREATE. 

If the value is zero, the expression before still creates the account with the empty codehash, but the proposed one doesn't.

If the value is NOT zero, the expression before still creates the account with the empty codehash, but the proposed one doesn't.

|A | B | C | Output|
|-- | -- | -- | --|
0 | 0 | 0 | F
0 | 0 | 1 | T
0 | 1 | 0 | F
0 | 1 | 1 | T
1 | 0 | 0 | F
1 | 0 | 1 | T
1 | 1 | 0 | T
1 | 1 | 1 | T





|A | B | C | Output|
|-- | -- | -- | --|
0 | 0 | 0 | F
0 | 0 | 1 | F
0 | 1 | 0 | F
0 | 1 | 1 | F
1 | 0 | 0 | F
1 | 0 | 1 | T
1 | 1 | 0 | T
1 | 1 | 1 | T



</p>
</details>